src/pages/guides/data-fetching.md

```diff
@@ -3,27 +3,62 @@ layout: ~/layouts/Main.astro
 title: Data Fetching
 ---
 
-Astro components and pages can fetch remote data to help generate your pages. Astro provides two different tools to pages to help you do this: **fetch()** and **top-level await.** 
+Astro components and pages can fetch remote data to help generate your pages. Astro provides two different tools to pages to help you do this: **fetch()** and **top-level await.**
 
-### `fetch()`
+## `fetch()`
 
-Astro pages have access to the global `fetch()` function in their setup script. `fetch()` is a native JavaScript API ([MDN](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch)) that lets you make HTTP requests for things like APIs and resources. 
+Astro pages have access to the global `fetch()` function in their setup script. `fetch()` is a native JavaScript API ([MDN](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch)) that lets you make HTTP requests for things like APIs and resources.
 
 Even though Astro component scripts run inside of Node.js (and not in the browser) Astro provides this native API so that you can fetch data at page build time.
 
-
 ```astro
+// Movies.astro
```

Comment:
nit: JS needs to be within the code fence, would break if copy-pasted:

```
---
// Movies.astro
```

Reply:
What if I use a HTML comment? I'd rather keep the comment at the top, though I imagine Astro parsing just breaks if the TS codeblock isn't at the top 🤔